benefits/core/session.py

```diff
@@ -157,10 +157,20 @@ def oauth_token(request):
 
 
 def oauth_claims(request):
-    """Get the oauth claim from the request's session, or None"""
+    """Get the oauth claims from the request's session, or None"""
     return request.session.get(_OAUTH_CLAIMS)
 
 
+def oauth_extra_claims(request, eligibility_claim, oauth_claims):
+    """Get the extra oauth claims from the request's session, or None"""
+    if oauth_claims is not None:
+        # since .remove works in-place, make a copy
+        all_claims = oauth_claims[:]
+        return all_claims.remove(eligibility_claim)
+    else:
+        return None
```

Comment:
This is starting to feel a little weird. As written, this doesn't feel like it belongs in `session`.

If it is in `session`, it should just use the other helpers here to get the data. E.g. `.flow()` to get the `eligibility_claim` and `.oauth_claims()` to get those.

Reply:
To be clear, I think it's fine to leave this in `session` -- but in that case, actually make use of the session! The `request` parameter isn't even used here.

Reply:
I agree, I ended up simplifying `oauth_extra_claims()` and realized that `request` is not used, so it feels like it doesn't belong in `session`. As I wrote it out, I realized that what this function needs (`eligibility_claim` and `oauth_claims`) should be passed into it because they are already available in (typically) the view that calls it. Then, the function is only taking out `eligibility_claim` from `oauth_claims`. 

If this makes sense, I'm struggling a bit to decide where `oauth_extra_claims()` would be declared.

Reply:
I think leave it in `session` since it is calculated from values in the session. 

But actually calculate it, don't pass anything in. This is similar to e.g. how the reenrollment date is calculated: https://github.com/cal-itp/benefits/blob/main/benefits/core/session.py#L107